magento/magento2
# Fixed creating shipping labels in part-shipment

Fixed creating package when we ship not all items


### Description (*)
When we use sources, we can create shipment for only one item from order. In this case when we try to create shipping label with shipment ("Create Shipping Label" Checkbox), the system requires to add all products to package (including products, which wasn't added to shipment) and validation fails.


### Related Pull Requests


### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

0. Have configured shipping method with ability to create Shipping Labels
1. Have 2 products with available in 2 or more sources
2. Add to cart and order both products with shipping method from point 0.
3. Create shipment from admin panel: select source and enter qty only for one product (qty for second product should be 0)
4. Click Proceed to Shipment.
5. Check that only one product in Items to Ship
6. Check checkbox "Create Shipping Label" and press "Submit Shipment..."
7. Click "Add Products to Package". In the list there are both products.
8. When we add to package only one product - the next button is disabled.

Expected result:
7. In the list there is only one product
8. The "Next" button is enabled

### Questions or comments


### Contribution checklist (*)
 - [*] Pull request has a meaningful description of its purpose
 - [*] All commits are accompanied by meaningful commit messages
 - [*] All new or changed code is covered with unit/integration tests (if applicable)
 - [*] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29552: Fixed creating shipping labels in part-shipment
